src/Search_Replace_Command.php

```diff
@@ -137,6 +137,12 @@ public function __invoke( $args, $assoc_args ) {
 		$this->regex_flags     =  \WP_CLI\Utils\get_flag_value( $assoc_args, 'regex-flags' );
 		$this->format          = \WP_CLI\Utils\get_flag_value( $assoc_args, 'format' );
 
+		// http://php.net/manual/en/reference.pcre.pattern.modifiers.php
+		if ( $this->regex_flags && ! preg_match( '/^[imsxeADSUXJu]+$/', $this->regex_flags ) ) {
```

Comment:
Should we maybe make sure that there are no duplicates switches either?

We can use a regex like this:
```
^(?!.*(.).*\1)[imsxeADSUXJu]+$
```